src/TorchSharp/NN/Utils/RNNUtils.cs

```diff
@@ -0,0 +1,90 @@
+                    /// <param name="lengths">The lengths of sequences</param>
+                    /// <param name="batch_first">If true, the size of input tensor is (batch, seq, *), otherwise (seq, batch, *)</param>
+                    /// <param name="enforce_sorted">if true, checks that the input contains sequences sorted by length in a decreasing order.</param>
+                    /// <returns></returns>
+                    public static PackedSequence pack_padded_sequence(torch.Tensor input, torch.Tensor lengths, bool batch_first = false, bool enforce_sorted = true)
+                    {
+                        var res = THSNN_pack_padded_sequence(input.Handle, lengths.Handle, batch_first, enforce_sorted);
+                        return new PackedSequence(res);
+                    }
+
+                    /// <summary>
+                    /// Pad a packed batch of variable length sequences.
+                    /// </summary>
+                    /// <param name="sequence">A packed batch</param>
+                    /// <param name="batch_first">If true, the size of output tensor is (batch, seq, *), otherwise (seq, batch, *)</param>
+                    /// <param name="padding_value">The values of padding.</param>
+                    /// <param name="total_length">If not null, the output tensor is padded to the length of total_length.</param>
+                    /// <returns>The padded tensor</returns>
+                    public static (torch.Tensor, torch.Tensor) pad_packed_sequence(PackedSequence sequence, bool batch_first = false, double padding_value = 0.0, long? total_length = null)
+                    {
+                        IntPtr res1, res2;
+                        long total_length_arg = total_length.HasValue ? total_length.Value : -1;
+                        THSNN_pad_packed_sequence(sequence.Handle, batch_first, padding_value, total_length_arg, out res1, out res2);
+                        return (new torch.Tensor(res1), new torch.Tensor(res2));
+                    }
+
+                    /// <summary>
+                    /// Pad a list of variable length sequences.
+                    /// </summary>
+                    /// <param name="sequences">A list of variable length sequences</param>
+                    /// <param name="batch_first">If true, the size of output tensor is (batch, seq, *), otherwise (seq, batch, *)</param>
+                    /// <param name="padding_value">The values of padding.</param>
+                    /// <returns>The padded tensor</returns>
+                    public static torch.Tensor pad_sequence(IEnumerable<torch.Tensor> sequences, bool batch_first = false, double padding_value = 0.0)
+                    {
+                        var sequences_arg = sequences.Select(p => p.Handle).ToArray();
+                        var res = THSNN_pad_sequence(sequences_arg, sequences_arg.Length, batch_first, padding_value);
+                        return new torch.Tensor(res);
```

Comment:
>res [](http://example.com/codeflow?start=48&length=3)

who is freeing these resource?

Reply:
I think `res` is owned by `Tensor` and it is freed in `Tensor.Dispose(bool)`. The caller or `DisposeScope` are responsible to free it, otherwise `~Tensor()` will do.